src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,72 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+
+        Rectangle current = (Rectangle) rectangle;
+
+        if (rectangle == this) {
+            return true;
+        }
+
+        if (rectangle == null) {
+            return false;
+        }
+
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            return colorCheck(current) && lengthCheck(current) && widthCheck(current);
```

Comment:
let's remove all those private methods and create a single statement using && and ||

follow the following logical flow for each field:

return true,

if both `field` and `rectangle.field` are null
OR if `field` is not null and equals `rectangle.field`